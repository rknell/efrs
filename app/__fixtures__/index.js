import users from './users.fixtures'

export default {
  users
}
