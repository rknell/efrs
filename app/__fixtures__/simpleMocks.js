
export class MockFormController {
  model = {
    disableScroll: false
  }
}
