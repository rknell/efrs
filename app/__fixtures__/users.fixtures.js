function ObjectId (data) {
  return data
}

function ISODate (data) {
  return data
}

export default {
  setupUser: {
    '_id': ObjectId('5a36fa6aac023a4b3bae2f22'),
    'createdAt': ISODate('2017-12-17T23:14:50.373Z'),
    'updatedAt': ISODate('2017-12-17T23:14:50.390Z'),
    'password': '$2a$08$gtSMlrmdhH0zOeeYJzPh.egq2I65fD9WFV5PypJfNWqikrZOzFQJq',
    'username': 'user12@example.com',
    'email': 'user12@example.com',
    'domain': 'example.com.au',
    'firstName': 'Admin',
    'lastName': 'User',
    'relationships': [],
    'pushTokens': [],
    'isSetup': true,
    '__v': 0,
    'company': ObjectId('5a36fa6aac023a4b3bae2f20')
  },
  newUser: {
    '_id': ObjectId('5a36fa6aac023a4b3bae2f22'),
    'createdAt': ISODate('2017-12-17T23:14:50.373Z'),
    'updatedAt': ISODate('2017-12-17T23:14:50.390Z'),
    'password': '$2a$08$gtSMlrmdhH0zOeeYJzPh.egq2I65fD9WFV5PypJfNWqikrZOzFQJq',
    'username': 'user12@example.com',
    'email': 'user12@example.com',
    'domain': 'example.com.au',
    'firstName': 'Admin',
    'lastName': 'User',
    'relationships': [],
    'pushTokens': [],
    'isSetup': false,
    '__v': 0,
    'company': ObjectId('5a36fa6aac023a4b3bae2f20')
  }
}
