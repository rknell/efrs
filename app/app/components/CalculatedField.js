import React, { Component } from 'react'
import { Text, View } from 'react-native'
import { observer } from 'mobx-react/native'
import EditModeWrapper from './EditModeWrapper'
import FieldController from '../services/FieldController'
import FieldBase from '../componentViews/FieldBase'
import GlobalStyle from '../services/GlobalStyle'

@observer
class CalculatedField extends Component {
  constructor (props) {
    super(props)
    this.controller = new CalculatedFieldController(props)
    this.onPressEdit = this.onPressEdit.bind(this)
  }

  componentWillMount () {
    this.controller.initialise()
  }

  onPressEdit () {
    this.controller.viewSettings()
  }

  render () {
    if (this.props.isSearching) {
      return null
    } else {
      return (<EditModeWrapper {...this.props} onPressEdit={this.onPressEdit}>
        <FieldBase hidden={this.props.editMode ? false : this.controller.settingsObject['Hidden']}
          globalStyle={GlobalStyle.style} label={this.controller.settingsObject.Label}>
          <View style={{ flex: 1, justifyContent: 'center' }}>
            <Text
              style={{ fontSize: 16 }}>{this.props.data.values.default !== 'undefined' ? this.props.data.values.default : ''}</Text>
          </View>
        </FieldBase>
      </EditModeWrapper>)
    }
  }
}

class CalculatedFieldController extends FieldController {
  settings = [
    {
      value: '',
      label: 'Label',
      fieldType: 'TextFieldView',
      extraProps: {}
    }, {
      value: '',
      label: 'Calculation',
      fieldType: 'TextFieldView',
      extraProps: {
        settings: { 'Multi Line': true, 'Auto Capitalize': 'none' }
      }
    }, {
      value: false,
      label: 'Hidden',
      fieldType: 'QuickPickView',
      extraProps: {
        options: [{ label: 'Yes', value: true }, { label: 'No', value: false }]
      }
    }, {
      value: false,
      label: 'Report When Hidden',
      fieldType: 'QuickPickView',
      extraProps: {
        options: [{ label: 'Yes', value: true }, { label: 'No', value: false }]
      }
    }
  ]
}
export default CalculatedField
