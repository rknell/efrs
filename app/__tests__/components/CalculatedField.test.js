import React from 'react'
import CalculatedField from '../../app/components/CalculatedField'
import { shallow } from 'enzyme'
import toJSON from 'enzyme-to-json'
import { MockFormController } from '../../__fixtures__/simpleMocks'

it('should match snapshot', () => {
  const wrapper = shallow(<CalculatedField formController={new MockFormController()} data={{
    values: {
      default: ''
    }
  }} />)
  expect(toJSON(wrapper)).toMatchSnapshot()
})

it('should match snapshot with some data', () => {
  const wrapper = shallow(<CalculatedField formController={new MockFormController()} data={{
    values: {
      default: 'test'
    }
  }} />)
  expect(toJSON(wrapper)).toMatchSnapshot()
})

it('should match snapshot with some data', () => {
  const wrapper = shallow(<CalculatedField formController={new MockFormController()} data={{
    values: {
      default: 'undefined'
    }
  }} />)
  expect(toJSON(wrapper)).toMatchSnapshot()
})

it('should render edit mode', () => {
  const wrapper = shallow(<CalculatedField formController={new MockFormController()} editMode data={{
    values: {
      default: 'test'
    }
  }} />)
  expect(toJSON(wrapper)).toMatchSnapshot()
})

it('should handle onPress edit', () => {
  const wrapper = shallow(<CalculatedField formController={new MockFormController()} data={{
    values: {
      default: ''
    },
    settings: []
  }} />)
  const instance = wrapper.instance()

  instance.onPressEdit()
})
