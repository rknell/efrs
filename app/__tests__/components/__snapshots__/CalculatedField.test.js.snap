// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<EditModeWrapper
  data={
    Object {
      "settings": Array [
        Object {
          "extraProps": Object {},
          "fieldType": "TextFieldView",
          "label": "Label",
          "value": "",
        },
        Object {
          "extraProps": Object {
            "settings": Object {
              "Auto Capitalize": "none",
              "Multi Line": true,
            },
          },
          "fieldType": "TextFieldView",
          "label": "Calculation",
          "value": "",
        },
        Object {
          "extraProps": Object {
            "options": Array [
              Object {
                "label": "Yes",
                "value": true,
              },
              Object {
                "label": "No",
                "value": false,
              },
            ],
          },
          "fieldType": "QuickPickView",
          "label": "Hidden",
          "value": false,
        },
        Object {
          "extraProps": Object {
            "options": Array [
              Object {
                "label": "Yes",
                "value": true,
              },
              Object {
                "label": "No",
                "value": false,
              },
            ],
          },
          "fieldType": "QuickPickView",
          "label": "Report When Hidden",
          "value": false,
        },
      ],
      "values": Object {
        "default": "",
      },
    }
  }
  formController={
    MockFormController {
      "model": Object {
        "disableScroll": true,
      },
    }
  }
  onPressEdit={[Function]}
>
  <FieldBase
    globalStyle={
      Object {
        "disabledText": "rgb(179, 179, 179)",
        "errorColour": "red",
        "fullLogo": Object {
          "testUri": "../../../app/assets/efrsFullBanner.png",
        },
        "neutralColour": "white",
        "neutralColourDisabled": "rgba(255, 255, 255, 0.3)",
        "neutralColourHighlight": "rgb(217, 217, 217)",
        "neutralColourIsDark": false,
        "neutralColourOffset": "rgb(242, 242, 242)",
        "neutralColourText": "rgb(51, 51, 51)",
        "neutralColourTextDisabled": "rgba(51, 51, 51, 0.3)",
        "primaryColour": "dodgerblue",
        "primaryColourDisabled": "rgba(30, 144, 255, 0.3)",
        "primaryColourHighlight": "rgb(106, 182, 255)",
        "primaryColourIsDark": true,
        "primaryColourText": "rgb(255, 255, 255)",
        "primaryColourTextDisabled": "rgba(255, 255, 255, 0.3)",
        "secondaryColour": "#f3f3f3",
        "secondaryColourDisabled": "rgba(243, 243, 243, 0.3)",
        "secondaryColourHighlight": "rgb(205, 205, 205)",
        "secondaryColourIsDark": undefined,
        "secondaryColourText": "rgb(39, 39, 39)",
        "secondaryColourTextDisabled": "rgba(39, 39, 39, 0.3)",
        "secondaryLogo": Object {
          "testUri": "../../../app/assets/icon.png",
        },
      }
    }
    hidden={false}
    label=""
  >
    <View
      style={
        Object {
          "flex": 1,
          "justifyContent": "center",
        }
      }
    >
      <Text
        style={
          Object {
            "fontSize": 16,
          }
        }
      />
    </View>
  </FieldBase>
</EditModeWrapper>
`;

exports[`should match snapshot with some data 1`] = `
<EditModeWrapper
  data={
    Object {
      "settings": Array [
        Object {
          "extraProps": Object {},
          "fieldType": "TextFieldView",
          "label": "Label",
          "value": "",
        },
        Object {
          "extraProps": Object {
            "settings": Object {
              "Auto Capitalize": "none",
              "Multi Line": true,
            },
          },
          "fieldType": "TextFieldView",
          "label": "Calculation",
          "value": "",
        },
        Object {
          "extraProps": Object {
            "options": Array [
              Object {
                "label": "Yes",
                "value": true,
              },
              Object {
                "label": "No",
                "value": false,
              },
            ],
          },
          "fieldType": "QuickPickView",
          "label": "Hidden",
          "value": false,
        },
        Object {
          "extraProps": Object {
            "options": Array [
              Object {
                "label": "Yes",
                "value": true,
              },
              Object {
                "label": "No",
                "value": false,
              },
            ],
          },
          "fieldType": "QuickPickView",
          "label": "Report When Hidden",
          "value": false,
        },
      ],
      "values": Object {
        "default": "test",
      },
    }
  }
  formController={
    MockFormController {
      "model": Object {
        "disableScroll": true,
      },
    }
  }
  onPressEdit={[Function]}
>
  <FieldBase
    globalStyle={
      Object {
        "disabledText": "rgb(179, 179, 179)",
        "errorColour": "red",
        "fullLogo": Object {
          "testUri": "../../../app/assets/efrsFullBanner.png",
        },
        "neutralColour": "white",
        "neutralColourDisabled": "rgba(255, 255, 255, 0.3)",
        "neutralColourHighlight": "rgb(217, 217, 217)",
        "neutralColourIsDark": false,
        "neutralColourOffset": "rgb(242, 242, 242)",
        "neutralColourText": "rgb(51, 51, 51)",
        "neutralColourTextDisabled": "rgba(51, 51, 51, 0.3)",
        "primaryColour": "dodgerblue",
        "primaryColourDisabled": "rgba(30, 144, 255, 0.3)",
        "primaryColourHighlight": "rgb(106, 182, 255)",
        "primaryColourIsDark": true,
        "primaryColourText": "rgb(255, 255, 255)",
        "primaryColourTextDisabled": "rgba(255, 255, 255, 0.3)",
        "secondaryColour": "#f3f3f3",
        "secondaryColourDisabled": "rgba(243, 243, 243, 0.3)",
        "secondaryColourHighlight": "rgb(205, 205, 205)",
        "secondaryColourIsDark": undefined,
        "secondaryColourText": "rgb(39, 39, 39)",
        "secondaryColourTextDisabled": "rgba(39, 39, 39, 0.3)",
        "secondaryLogo": Object {
          "testUri": "../../../app/assets/icon.png",
        },
      }
    }
    hidden={false}
    label=""
  >
    <View
      style={
        Object {
          "flex": 1,
          "justifyContent": "center",
        }
      }
    >
      <Text
        style={
          Object {
            "fontSize": 16,
          }
        }
      >
        test
      </Text>
    </View>
  </FieldBase>
</EditModeWrapper>
`;

exports[`should match snapshot with some data 2`] = `
<EditModeWrapper
  data={
    Object {
      "settings": Array [
        Object {
          "extraProps": Object {},
          "fieldType": "TextFieldView",
          "label": "Label",
          "value": "",
        },
        Object {
          "extraProps": Object {
            "settings": Object {
              "Auto Capitalize": "none",
              "Multi Line": true,
            },
          },
          "fieldType": "TextFieldView",
          "label": "Calculation",
          "value": "",
        },
        Object {
          "extraProps": Object {
            "options": Array [
              Object {
                "label": "Yes",
                "value": true,
              },
              Object {
                "label": "No",
                "value": false,
              },
            ],
          },
          "fieldType": "QuickPickView",
          "label": "Hidden",
          "value": false,
        },
        Object {
          "extraProps": Object {
            "options": Array [
              Object {
                "label": "Yes",
                "value": true,
              },
              Object {
                "label": "No",
                "value": false,
              },
            ],
          },
          "fieldType": "QuickPickView",
          "label": "Report When Hidden",
          "value": false,
        },
      ],
      "values": Object {
        "default": "undefined",
      },
    }
  }
  formController={
    MockFormController {
      "model": Object {
        "disableScroll": true,
      },
    }
  }
  onPressEdit={[Function]}
>
  <FieldBase
    globalStyle={
      Object {
        "disabledText": "rgb(179, 179, 179)",
        "errorColour": "red",
        "fullLogo": Object {
          "testUri": "../../../app/assets/efrsFullBanner.png",
        },
        "neutralColour": "white",
        "neutralColourDisabled": "rgba(255, 255, 255, 0.3)",
        "neutralColourHighlight": "rgb(217, 217, 217)",
        "neutralColourIsDark": false,
        "neutralColourOffset": "rgb(242, 242, 242)",
        "neutralColourText": "rgb(51, 51, 51)",
        "neutralColourTextDisabled": "rgba(51, 51, 51, 0.3)",
        "primaryColour": "dodgerblue",
        "primaryColourDisabled": "rgba(30, 144, 255, 0.3)",
        "primaryColourHighlight": "rgb(106, 182, 255)",
        "primaryColourIsDark": true,
        "primaryColourText": "rgb(255, 255, 255)",
        "primaryColourTextDisabled": "rgba(255, 255, 255, 0.3)",
        "secondaryColour": "#f3f3f3",
        "secondaryColourDisabled": "rgba(243, 243, 243, 0.3)",
        "secondaryColourHighlight": "rgb(205, 205, 205)",
        "secondaryColourIsDark": undefined,
        "secondaryColourText": "rgb(39, 39, 39)",
        "secondaryColourTextDisabled": "rgba(39, 39, 39, 0.3)",
        "secondaryLogo": Object {
          "testUri": "../../../app/assets/icon.png",
        },
      }
    }
    hidden={false}
    label=""
  >
    <View
      style={
        Object {
          "flex": 1,
          "justifyContent": "center",
        }
      }
    >
      <Text
        style={
          Object {
            "fontSize": 16,
          }
        }
      />
    </View>
  </FieldBase>
</EditModeWrapper>
`;

exports[`should render edit mode 1`] = `
<EditModeWrapper
  data={
    Object {
      "settings": Array [
        Object {
          "extraProps": Object {},
          "fieldType": "TextFieldView",
          "label": "Label",
          "value": "",
        },
        Object {
          "extraProps": Object {
            "settings": Object {
              "Auto Capitalize": "none",
              "Multi Line": true,
            },
          },
          "fieldType": "TextFieldView",
          "label": "Calculation",
          "value": "",
        },
        Object {
          "extraProps": Object {
            "options": Array [
              Object {
                "label": "Yes",
                "value": true,
              },
              Object {
                "label": "No",
                "value": false,
              },
            ],
          },
          "fieldType": "QuickPickView",
          "label": "Hidden",
          "value": false,
        },
        Object {
          "extraProps": Object {
            "options": Array [
              Object {
                "label": "Yes",
                "value": true,
              },
              Object {
                "label": "No",
                "value": false,
              },
            ],
          },
          "fieldType": "QuickPickView",
          "label": "Report When Hidden",
          "value": false,
        },
      ],
      "values": Object {
        "default": "test",
      },
    }
  }
  editMode={true}
  formController={
    MockFormController {
      "model": Object {
        "disableScroll": true,
      },
    }
  }
  onPressEdit={[Function]}
>
  <FieldBase
    globalStyle={
      Object {
        "disabledText": "rgb(179, 179, 179)",
        "errorColour": "red",
        "fullLogo": Object {
          "testUri": "../../../app/assets/efrsFullBanner.png",
        },
        "neutralColour": "white",
        "neutralColourDisabled": "rgba(255, 255, 255, 0.3)",
        "neutralColourHighlight": "rgb(217, 217, 217)",
        "neutralColourIsDark": false,
        "neutralColourOffset": "rgb(242, 242, 242)",
        "neutralColourText": "rgb(51, 51, 51)",
        "neutralColourTextDisabled": "rgba(51, 51, 51, 0.3)",
        "primaryColour": "dodgerblue",
        "primaryColourDisabled": "rgba(30, 144, 255, 0.3)",
        "primaryColourHighlight": "rgb(106, 182, 255)",
        "primaryColourIsDark": true,
        "primaryColourText": "rgb(255, 255, 255)",
        "primaryColourTextDisabled": "rgba(255, 255, 255, 0.3)",
        "secondaryColour": "#f3f3f3",
        "secondaryColourDisabled": "rgba(243, 243, 243, 0.3)",
        "secondaryColourHighlight": "rgb(205, 205, 205)",
        "secondaryColourIsDark": undefined,
        "secondaryColourText": "rgb(39, 39, 39)",
        "secondaryColourTextDisabled": "rgba(39, 39, 39, 0.3)",
        "secondaryLogo": Object {
          "testUri": "../../../app/assets/icon.png",
        },
      }
    }
    hidden={false}
    label=""
  >
    <View
      style={
        Object {
          "flex": 1,
          "justifyContent": "center",
        }
      }
    >
      <Text
        style={
          Object {
            "fontSize": 16,
          }
        }
      >
        test
      </Text>
    </View>
  </FieldBase>
</EditModeWrapper>
`;
